tslint.json

```diff
@@ -5,8 +5,9 @@
   ],
   "rules": {
     "ban-ts-ignore": true,
-    "quotemark": [true, "single"],
-    "eofline": true
+    "curly": true,
```

Comment:
Couldn't find `curly` in internal TS Style guide. I feel like this style rule has some exceptions. 😀

Reply:
Internal TS and JS style guides both allow for single-line no-braces when it fits, which I think would be `"curly": [true, "ignore-same-line"]`.

Personally I feel like allowing exceptions immediately makes practices less straightforward and code less readable.

I think we're kind of 50/50 in terms of committers advocating that vs. advocating always-braces... @acdvorak what do you think about permitting this?

Reply:
I think we should always require `return` statements to be on their own line, surrounded by curly braces. No exceptions.

Working in Search Infrastructure taught me that brevity does not always equal clarity. I prefer code that is unambiguous and impossible to misinterpret over code that is easy to write.

Before I came to Google, I wrote same-line-if-returns all over the place. But after spending several years working deep in the bowels of [GWS](https://en.wikipedia.org/wiki/Google_Web_Server)—one of the largest, most complicated code bases on the planet, collectively written and maintained by tens of thousands of engineers over two decades—I've learned to appreciate the wisdom of consistent return statement syntax.

When you have a lot of engineers working on the same code ("a lot" meaning 10,000, not ~10), consistent and predictable code formatting, coupled with strict readability guidelines enforced by automatic linters and nitpicky coworkers, can make a big difference in maintainability.

Rules like "return statements should always be on their own line surrounded by curly braces" may seem silly or draconian, but they optimize scannability for the greatest number of engineers, especially internally.

I'm not convinced that allowing `return` statements on the same line as their condition _improves_ readability over curly braces. The internal style guide might allow it as an exception case, but that doesn't mean we have to.

For the record, I totally understand the perspective of preferring terser code. I came from a jquery/lodash/unix world where every API name was 6 chars or less (and did 50 different things depending on how you called it). Google's internal JS style guide (and `goog.some.really.long.namespace.Function` BS) annoyed me at first, but I've seen their value firsthand.

`<img src="change-my-mind.jpg">`